utils/resultstable_test.go

```diff
@@ -83,26 +83,154 @@ func TestGetDirectComponents(t *testing.T) {
+			{ComponentId: "gav://jfrogpack:1.0.0", FullPath: "jfrogpack:-1.0.0.jar"}},
+		{{ComponentId: "build://dort:1"},
+			{ComponentId: "gav://jfrogpack:1.0.0", FullPath: "jfrogpack:-1.0.0.jar"}},
+	}
+	expectedImpactPathRootsTrue := [][]services.ImpactPathNode{
+		{{ComponentId: "build://dort:1"},
+			{ComponentId: "gav://jfrogpack:1.0.0", FullPath: "jfrogpack:-1.0.0.jar"}},
+	}
+
+	testSimplifyVulnerabilityRoot(t, vulnerabilities, false, expectedImpactPathRootsFalse)
+	testSimplifyVulnerabilityRoot(t, vulnerabilities, true, expectedImpactPathRootsTrue)
+
+}
+
+func testSimplifyVulnerabilityRoot(t *testing.T, vulnerabilities []services.Vulnerability, multipleRoots bool, expectedImpactPath [][]services.ImpactPathNode) {
+	simplifiedVulnerabilities := simplifyVulnerabilities(vulnerabilities, multipleRoots)
+	assert.Equal(t, len(vulnerabilities)-1, len(simplifiedVulnerabilities))
+	for _, vulnerability := range simplifiedVulnerabilities {
+		for key := range vulnerability.Components {
+			if key == "gav://jfrogpack:1.0.0" {
+				assert.Equal(t, expectedImpactPath, vulnerability.Components[key].ImpactPaths)
+			}
+		}
+	}
+}
+
+func TestSimplifyViolation(t *testing.T) {
+	violations := []services.Violation{
+		{Components: map[string]services.Component{"gav://jfrogpack:1.0.0": {ImpactPaths: [][]services.ImpactPathNode{
+			{{ComponentId: "build://dort:1"},
+				{ComponentId: "generic://sha256:1bcd6597181d476796e206e176ccc185b4709ff28fb069c42e7f7f67c6a0ff28/multi3-3.7-20240806.082023-11.war",
+					FullPath: "multi3-3.7-20240806.082023-11.war"},
+				{ComponentId: "gav://jfrogpack:1.0.0", FullPath: "jfrogpack:-1.0.0.jar"}},
+		}}}, FailBuild: true},
+		{Components: map[string]services.Component{"gav://jfrogpack:1.0.1": {}}, FailBuild: true},
+		{Components: map[string]services.Component{"gav://jfrogpack:1.0.0": {ImpactPaths: [][]services.ImpactPathNode{
+			{{ComponentId: "build://dort:1"},
+				{ComponentId: "gav://jfrogpack:1.0.0", FullPath: "jfrogpack:-1.0.0.jar"}},
+		}}}, FailBuild: true},
+		{Components: map[string]services.Component{"gav://jfrogpack:1.0.2": {}}, FailBuild: true},
+	}
+	expectedImpactPathRootsFalse := [][]services.ImpactPathNode{
+		{{ComponentId: "build://dort:1"},
+			{ComponentId: "generic://sha256:1bcd6597181d476796e206e176ccc185b4709ff28fb069c42e7f7f67c6a0ff28/multi3-3.7-20240806.082023-11.war",
+				FullPath: "multi3-3.7-20240806.082023-11.war"},
+			{ComponentId: "gav://jfrogpack:1.0.0", FullPath: "jfrogpack:-1.0.0.jar"}},
+		{{ComponentId: "build://dort:1"},
+			{ComponentId: "gav://jfrogpack:1.0.0", FullPath: "jfrogpack:-1.0.0.jar"}},
+	}
+	expectedImpactPathRootsTrue := [][]services.ImpactPathNode{
+		{{ComponentId: "build://dort:1"},
+			{ComponentId: "gav://jfrogpack:1.0.0", FullPath: "jfrogpack:-1.0.0.jar"}},
+	}
+
+	testSimplifyViolationRoot(t, violations, false, expectedImpactPathRootsFalse)
+	testSimplifyViolationRoot(t, violations, true, expectedImpactPathRootsTrue)
```

Comment:
```suggestion
        for _, test := range tests {
		t.Run(test.testName, func(t *testing.T) {
		        testSimplifyViolationRoot(t, violations, test.isMultiRoot, test.expectedImpactPathRoots)
		})
	}
```
Implement this with testCases, same as `TestGetOperationalRiskSimplifiedViolations`